### PR TITLE
Fixed hanging issues in pubSub unit test.

### PR DESCRIPTION
**Pull request by chevtche**

### CODE REVIEW

File: tests/pubSub.cpp [hunk 1]
```diff
@@ -152,49 +152,52 @@ void onLargeEcho( const zeq::Event& event )
...
+        while( subscriber->receive( 0 )) /* NOP to drain */;
     }
+
     const auto& echoTime = std::chrono::high_resolution_clock::now() -
                            startTime;
 
     BOOST_CHECK_MESSAGE( noEchoTime < echoTime,
                          std::chrono::nanoseconds( noEchoTime ).count() << ", "
                          << std::chrono::nanoseconds( echoTime ).count( ));
+delete publisher;
```
Comment (tribal-tec) [inline tests/pubSub.cpp]: indentation

File: tests/pubSub.cpp [hunk 2]
```diff
@@ -152,49 +152,52 @@ void onLargeEcho( const zeq::Event& event )
 
 BOOST_AUTO_TEST_CASE(test_publish_receive_filters)
 {
-    zeq::Publisher publisher( servus::URI( "foo://" ), zeq::ANNOUNCE_NONE );
-    zeq::Subscriber subscriber( publisher.getURI( ));
+    zeq::Publisher* publisher = new zeq::Publisher( servus::URI( "foo://" ), zeq::ANNOUNCE_NONE );
+    zeq::Subscriber* subscriber = new zeq::Subscriber( publisher->getURI( ));
```
Comment (tribal-tec) [inline tests/pubSub.cpp]: the subscriber can stay on the stack

**Comment by tribal-tec** [comment]
Maybe give more information what's happening. Basically what we found out while debugging it.

**Comment by chevtche** [comment]
Retest this please

**Comment by chevtche** [comment]
Retest this please

**Comment by chevtche** [comment]
Can this be merged?

**Pull request merged**